src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -77,7 +77,7 @@ private void setUpDataDir() {
   /**
    * Delete a file or directory.
    **/
-  private void deleteFile(File fileToDelete) {
+  protected void deleteFile(File fileToDelete) {
```

Comment:
rm -rf is one option, but it's nice to use Java API's. Yes, this correct removes symlinks each time. The method has been tested often for this.